src/ray/common/task/task_spec.cc

```diff
@@ -594,15 +594,13 @@ std::string TaskSpecification::CallSiteString() const {
 }
 
 WorkerCacheKey::WorkerCacheKey(
-    const std::string serialized_runtime_env,
+    std::string serialized_runtime_env,
     const absl::flat_hash_map<std::string, double> &required_resources,
     bool is_actor,
     bool is_gpu,
     bool is_root_detached_actor)
-    : serialized_runtime_env(serialized_runtime_env),
-      required_resources(RayConfig::instance().worker_resource_limits_enabled()
-                             ? required_resources
-                             : absl::flat_hash_map<std::string, double>{}),
+    : serialized_runtime_env(std::move(serialized_runtime_env)),
+      required_resources(required_resources),
```

Comment:
Required_resources can also be removed?

Reply:
I don't think it's always empty. It describes the resource shape of the task, populated from Python side to CoreWorker: https://github.com/ray-project/ray/blob/e208dc550cd5705115c13e4dd8a5abcfc5ff3900/src/ray/core_worker/core_worker.cc#L2552

Reply:
I mean `required_resources` of `WorkerCacheKey` is always empty if the flag is off?

Reply:
removed